autotest/cpp/test_gdal.cpp

```diff
@@ -2581,13 +2581,22 @@ TEST_F(test_gdal, GetRasterNoDataReplacementValue)
     EXPECT_EQ(GDALGetRasterNoDataReplacementValue(
                   GDT_Float32, -std::numeric_limits<double>::infinity()),
               0.0);
+
     // in range for float 32
     EXPECT_EQ(static_cast<float>(
                   GDALGetRasterNoDataReplacementValue(GDT_Float32, -1.0)),
               std::nextafter(float(-1.0), 0.0f));
     EXPECT_EQ(static_cast<float>(
                   GDALGetRasterNoDataReplacementValue(GDT_Float32, 1.1)),
               std::nextafter(float(1.1), 2.0f));
+    EXPECT_EQ(GDALGetRasterNoDataReplacementValue(
```

Comment:
you perhaps need to static_cast<float> the return of  GDALGetRasterNoDataReplacementValue?

Reply:
ah, ignore my above remark. Seems you've addressed it already